### ISSUE TITLE
polarity of `queryParamExplode` is backwards in `serializeQueryParam`

### ISSUE THREAD

**Issue by stepcut** [Haskell-OpenAPI-Code-Generator/Haskell-OpenAPI-Client-Code-Generator#74]
I am pretty sure the polarity for `queryParamExplode` is inverted. This patch should fix it:

https://github.com/Haskell-OpenAPI-Code-Generator/Haskell-OpenAPI-Client-Code-Generator/commit/534b45874b6035779dda47af6e0081baacd7842c

If I understand the code correctly, it would seem that `jsonToFormDataFlat` generates the query parameters in the exploded form. So `queryParamExplode` should do nothing (aka, `id`) if the parameters are supposed to be exploded. And it needs to consolidate them if they are not supposed to be exploded.

I had issues calling the mailchimp marketing api until I fixed this polarity in the generated code. When I printed out the request, the query parameters were exploded when the should not have been.

**Comment by joelfisch** [comment]
Hey @stepcut, thank you for noticing, you are absolutely right!

I noticed while investigating that there are other flaws as well with the current implementation and I fixed them as well as added tests to make sure it should work in the future too in this [commit](https://github.com/Haskell-OpenAPI-Code-Generator/Haskell-OpenAPI-Client-Code-Generator/commit/cddc4f40e02c1b130fc1c405952d05e072d694ff). I will create a PR for it shortly and after merging close this issue if this is fine with you.
